Code/Tools/ProjectManager/Source/CreateAGemScreen.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <ScreenWidget.h>
+#include <FormFolderBrowseEditWidget.h>
+#include <FormLineEditWidget.h>
+
+#include <QApplication>
+#include <QStyleOptionTab>
+#include <QStylePainter>
+#include <QTabBar>
+#include <QTabWidget>
+#include <QDialogButtonBox>
+#include <QPushButton>
+#include <QScrollArea>
+
+
+#endif
+
+class TabBar : public QTabBar
+{
+public:
+    QSize tabSizeHint(int index) const
+    {
+        QSize s = QTabBar::tabSizeHint(index);
+        s.transpose();
+        return s;
+    }
+
+protected:
+    void paintEvent(QPaintEvent* /*event*/)
+    {
+        QStylePainter painter(this);
+        QStyleOptionTab opt;
+
+        QPoint currentTopPosition(115, 175);
+
+        QStringList strs = { "1. Gem Setup", "2. Gem Details", "3. Creator Details" };
+        for (int i = 0; i < count(); i++)
+        {
+            initStyleOption(&opt, i);
+            painter.drawControl(QStyle::CE_TabBarTabShape, opt);
+            painter.save();
+            QSize s = opt.rect.size();
+            s.transpose();
+            QRect r(QPoint(), s);
+            r.moveCenter(opt.rect.center());
+            opt.rect = r;
+            QPoint c = tabRect(i).center();
+            QPoint leftJustify(currentTopPosition);
+            leftJustify.setX(30 + (int)(0.5 * opt.rect.width()));
+            painter.translate(leftJustify);
+            currentTopPosition.setY(currentTopPosition.y() + 55);
+            //painter.rotate(90);
+            painter.translate(-c);
+            // painter.drawControl(QStyle::CE_TabBarTabLabel, opt);
+            painter.drawItemText(r, Qt::AlignLeft, QApplication::palette(), true, strs.at(i));
+            painter.restore();
+        }
+    }
+};
+
+class TabWidget : public QTabWidget
+{
+public:
+    TabWidget(QWidget* parent = 0)
+        : QTabWidget(parent)
+    {
+        setTabBar(new TabBar);
+        setTabPosition(QTabWidget::West);
+    }
+};
```

Comment:
Move this to the .cpp file or create a new .h & .cpp if this approach really is necessary.  Including unnecessary code in a header typically leads to slower compile times due to it getting unintentionally included by other files.  It's best to get in the habit of keeping your headers as lean as possible to keep compile times down and make them easier to read.

Reply:
Leaving as is for this pull request as adding new files will probably require a full rebuild.